src/model.py

```diff
@@ -483,6 +512,13 @@ def finalize(self):
         """
         self.logger.log_info("[vllm] Issuing finalize to vllm backend")
         self._shutdown_event.set()
+
+        # Signal shutdown to the response sender thread.
+        self._response_queue.put(None)
```

Comment:
is `put` here ok, or should we use `put_nowait` here as well?

Reply:
I think it is ok, because we are unloading the model instance.